src/common/commonutils/CommandUtils.c

```diff
@@ -29,270 +31,194 @@ static int NormalizeStatus(int status)
-        if (0 == (workerProcess = fork()))
-        {
-            // Worker process
-            status = execl("/bin/sh", "sh", "-c", command, (char*)NULL);
-            _exit(status);
-        }
-        else if (workerProcess > 0)
-        {
-            // Wait on the worker process to terminate
-            waitpid(workerProcess, &status, 0);
-        }
-        else
-        {
-            // If our fork fails, try system(), if that also fails then the call fails
-            OsConfigLogError(log, "Failed forking process to execute command (errno: %d, '%s'), attempting system", errno, strerror(errno));
-            status = system(command);
-        }
-    }
-
-    status = NormalizeStatus(status);
-    OsConfigLogDebug(log, "SystemCommand: command '%s' returning %d", command, status);
-
-    return status;
-}
-
-#define MAX_COMMAND_RESULT_FILE_NAME 100
-
-int ExecuteCommand(void* context, const char* command, bool replaceEol, bool forJson, unsigned int maxTextResultBytes, unsigned int timeoutSeconds, char** textResult, CommandCallback callback, OsConfigLogHandle log)
-{
-    const char commandTextResultFileTemplate[] = "/tmp/~OSConfig.TextResult%u";
-    const char commandSeparator[] = " > ";
-    const char commandTerminator[] = " 2>&1";
-
-    int status = -1;
-    FILE* resultsFile = NULL;
-    int fileSize = 0;
-    int next = 0;
-    int i = 0;
-    char* commandLine = NULL;
-    size_t commandLineLength = 0;
-    size_t maximumCommandLine = 0;
-    char commandTextResultFile[MAX_COMMAND_RESULT_FILE_NAME] = {0};
-    bool wrappedCommand = false;
-
-    if ((NULL == command) || (0 == system(NULL)))
-    {
-        OsConfigLogDebug(log, "Cannot run command '%s'", command);
-        return -1;
-    }
-
-    commandLineLength = strlen(command);
-    wrappedCommand = (('(' == command[0]) || (')' == command[commandLineLength]));
-
-    // Append a random number to the results file to prevent parallel commands overwriting each other results
-    snprintf(commandTextResultFile, sizeof(commandTextResultFile), commandTextResultFileTemplate, rand());
-
-    commandLineLength += strlen(commandSeparator) + strlen(commandTextResultFile) + strlen(commandTerminator) + (wrappedCommand ? 0 : 2) + 1;
-
-    maximumCommandLine = (size_t)sysconf(_SC_ARG_MAX);
-    if (commandLineLength > maximumCommandLine)
-    {
-        OsConfigLogDebug(log, "Cannot run command '%s', command too long (%u), ARG_MAX: %u", command, (unsigned)commandLineLength, (unsigned)maximumCommandLine);
-        return E2BIG;
-    }
-
-    commandLine = (char*)malloc(commandLineLength);
-    if (NULL == commandLine)
-    {
-        OsConfigLogError(log, "Cannot run command '%s', cannot allocate %u bytes for command, out of memory", command, (unsigned)commandLineLength);
-        return ENOMEM;
-    }
-
-    snprintf(commandLine, commandLineLength, wrappedCommand ? "%s%s%s%s" : "(%s)%s%s%s", command, commandSeparator, commandTextResultFile, commandTerminator);
```

Comment:
>wrappedCommand [](http://example.com/codeflow?start=45&length=14)

Is this refactor (rework) keeping support for wrapped commands? Commands which by their syntax could interfere with our command execution command?

Reply:
Previously the commands were called in subshell (if they weren't wrapped in () by the caller, hence this check) to make sure that stdout/stderr redirection works in shell. That's no longer necessary.